tubular/imputers.py

```diff
@@ -90,40 +85,41 @@ def __init__(self, impute_value, columns, **kwargs):
 
     def transform(self, X):
         """Impute missing values with the supplied impute_value.
-
         If columns is None all columns in X will be imputed.
-
         Parameters
         ----------
         X : pd.DataFrame
             Data containing columns to impute.
-
         Returns
         -------
         X : pd.DataFrame
             Transformed input X with nulls imputed with the specified impute_value, for the specified columns.
 
+        Additions
+        ---------
+        * Preserving the datatypes of columns
+        * Finding the target column dtype and cast imputer values as same dtype
         """
 
-        self.check_is_fitted(["impute_value"])
+        self.check_is_fitted(["impute_values_"])
         self.columns_check(X)
 
         for c in self.columns:
+                
+                if "category" in X[c].dtype.name:
 
-            # for categorical dtypes have to set new category for the impute values first
-            if "category" in X[c].dtype.name:
+                    if self.impute_value not in X[c].cat.categories:
 
-                if self.impute_value not in X[c].cat.categories:
+                        X[c] = X[c].cat.add_categories(self.impute_value) # add new category
 
-                    X[c] = X[c].cat.add_categories(self.impute_value)
+                dtypes = X[c].dtype # preserve dtype
```

Comment:
`dtypes` variable only holds one dtype value at a time so the name is a little misleading.  Please change to something like `dtype` or `col_dtype`